scene/animation/animation_blend_tree.h

```diff
@@ -411,6 +411,7 @@ class AnimationNodeBlendTree : public AnimationRootNode {
 	};
 
 	RBMap<StringName, Node, StringName::AlphCompare> nodes;
+	RBMap<AnimationNode *, StringName> node_to_name;
```

Comment:
I believe this approach is redundant. At the very least, storing the same String in two places is not the right direction, and we must do indirect mapping, such as storing only the index, and the entity as a String must be only one.

Reply:
isn't StringName only storing the unified data pointer within either map, so there is just one backing string? this is just to avoid the linear cost of only having `nodes` to determine a name in the current context. ideally there would just be a bimap template available, but alternatively the name could also be copied to `NodeState` at the same time that the base path and parent are set.

Reply:
I am saying that the concatnated path to the AnimationNode already exists in the property_reference_map, so you don't need to create a new map and just retrieve it.

Reply:
Do you think it would be better to parse out the final part of the path or emit the full concatenated path?

Reply:
The signal must return the full path to the AnimationNode. As I have said many times, since the AnimationNode is a Resource, the same reference AnimationNodes can be placed in the same hierarchy. To distinguish between them, the user must know the full path.

Reply:
Ah yes that makes sense. I was only considering blend trees as roots but of course the names alone would break if they were nested.